# Predicting peptides from Ticks on a Tree & Trait Mapping outputs 

[![run with conda](http://img.shields.io/badge/run%20with-conda-3EB049?labelColor=000000&logo=anaconda)](https://docs.conda.io/projects/miniconda/en/latest/)
[![Snakemake](https://img.shields.io/badge/snakemake--green)](https://snakemake.readthedocs.io/en/stable/)

## Purpose

This repository predicts peptides from tick proteins that are associated with itch suppression.

## Installation and Setup

This repository uses Snakemake to run the pipeline and conda to manage software environments and installations. You can find operating system-specific instructions for installing miniconda [here](https://docs.conda.io/projects/miniconda/en/latest/). After installing conda and [mamba](https://mamba.readthedocs.io/en/latest/), run the following command to create the pipeline run environment.

```{bash}
mamba env create -n ticktides --file envs/dev.yml
conda activate ticktides
```

Snakemake manages rule-specific environments via the `conda` directive and using environment files in the [envs/](./envs/) directory. Snakemake itself is installed in the main development conda environment as specified in the [dev.yml](./envs/dev.yml) file.

To start the pipeline, run:

```{bash}
snakemake -s analyze-peptigate-outputs.snakefile --software-deployment-method conda -j 8
```

## Data

This repository analyzes the outputs of three previous analyses to predict peptide sequences from proteins that are associated with the itch-suppression trait.
The two previous analyses are:
1. [**protein-data-curation**](https://github.com/Arcadia-Science/protein-data-curation/): This repository provides a workflow to download, process, and annotate genomic and transcriptomic data that can then be fed to NovelTree (see next point). The input sequences are all from chelicerate species and are documented [here](https://github.com/Arcadia-Science/2023-chelicerate-analysis/blob/main/inputs/samples.tsv). 
1. **Ticks on a Tree**: [NovelTree](https://github.com/Arcadia-Science/noveltree) applied to the curated chelicerate data set. The input data set to NovelTree is protein sequences (either genome gene annotations or open reading frames predicted from transcriptomes). NovelTree applies evolutionary analyses to identify proteins that are "novel" under models of speciation, loss, or transfer. 
2. [**phylo-profiling**](https://github.com/Arcadia-Science/phylo-profiling/tree/main): trait mapping applied to the results of the ToT NovelTree run to identify proteins (from clusters of orthologous groups) that are associated with the itch-suppression trait.

The input to this repository is all proteins predicted to be associated with the itch-suppression trait.
These proteins are recorded in the file [`pos_speciation_proteins.fasta.gz`](./inputs/pos_speciation_proteins.fasta.gz).

We grabbed these proteins by:
1. In the [**phylo-profiling**](https://github.com/Arcadia-Science/phylo-profiling/tree/main) repository, there is a folder `chelicerate-results/itch_association_results/combined_profile_clusters/speciation_associations/`. Inside this folder are clusters of orthologous groups that were associated with itch suppression. These clusters are named `cluster_33`, `cluster_46`, `cluster_59`, `cluster_64`, `cluster_66`, and `cluster_71`. To reduce the number of statistical tests and the subsequent loss of power by p-value correction, this analysis grouped similar orthologs into clusters and performed itch suppression association analysis on those clusters. It then tries to identify which orthologous groups drive the association. None of the orthologous groups were associated with itch suppression on their own after FDR-correction. However, we chose to initially investigate all protein sequences from all orthologous groups in these clusters. The folder linked above contains TSV files that document the orthologous groups that make up each cluster.
2. To retrieve the protein sequences themselves, as well as metadata generated by the [protein-data-curation](https://github.com/Arcadia-Science/protein-data-curation) workflow, we used the scripts created by @elizabethmcd in the [phylo-profiling](https://github.com/Arcadia-Science/phylo-profiling/tree/main) repository. The first step of linking metadata files is documented in the file `scripts/clusters-orthogroups-analysis.R` (TER TODO: LINK TO SCRIPT WHEN MRG'D TO MAIN). While this script ends up doing rounds of filtering for @elizabethmcd's own analyses, we retained all metadata for all protein sequences as stated in the previous point ([`pos_speciation_signf_clusters_orthogroups_annotations.tsv.gz`](./inputs/pos_speciation_signf_clusters_orthogroups_annotations.tsv.gz)).
3. We then used the script `grab_proteins_from_locus_tags.py` (TER TODO LINK TO SCRIPT WHEN MRG'D TO MAIN) to extract the protein sequences themselves from the chelicerate data NovelTree run input files. The file `all_chelicerate_proteins.fasta` was created by concatenating the outputs of the `protein-data-curation` workflow (`s3://arcadia-github-repository-outputs/2023-chelicerate-analysis/25aacutoff_fullset_100223/processedprots/*fasta`). The protein sequences produced by the `grab_proteins_from_locus_tags.py` are the inputs into the analyses in this repository.
```{bash}
python3 grab_proteins_from_locus_tags.py all_chelicerate_proteins.fasta pos_speciation_locus_tags.txt pos_speciation_proteins.fasta
gzip pos_speciation_proteins.fasta
```

## Overview

Using the sequences of proteins predicted to be associated with the itch-suppression trait, we predicted peptide sequences from these proteins.
We define peptides as any protein sequences of less than or equal to 100 amino acids in length.

The peptigate pipeline predicted peptides from transcriptome assemblies.
It had two prediction modules that targetted small open reading frames (sORFs) and cleavage peptides, respectively.
Because we had access to protein sequences but not to transcriptome assemblies, we modified our peptide prediction approach to retrieve as many peptides as possible given our input data.
We:
1. **sORF prediction**: filtered the itch-suppression-associated proteins to those that were 100 amino acids or less. This was where we suffered the most potential loss in our predictive capabilities -- we think we would have many more potential sORF sequences if we had transcriptomes as input instead of protein sequences. However, these would require a new trait mapping analysis.
2. **Cleavage peptide prediction**: ran the cleavage peptide prediction portion of the peptigate pipeline on protein sequences longer than 100 amino acids.
3. **Peptide annotation**: annotated the predicted peptides with the annotation and analysis portion of the peptigate pipeline.

We ran peptigate from commit `4ecd7d5d02ce9ff3216f47babf76466717105d4e` with the following command:

```
snakemake -s protein_as_input.snakefile --software-deployment-method conda -j 1 -k --configfile input_configs/tot_protein_peptigate_config.yml
```

where the `tot_protein_peptigate_config.yml` looked like:
```
input_dir: "inputs/"
output_dir: "outputs/ToT"
orfs_amino_acids: "input_data/ToT/pos_speciation_proteins.fasta"
```

With these peptide predictions, we then proceeded with the rest of the analysis.

We ran additional annotation and comparison analysis with the snakefile in this repository:

```
snakemake -s analyze-peptigate-outputs.snakefile --software-deployment-method conda -j 8
```

This snakefile:
* Compares (BLASTp) peptide predictions against peptides in the Human Peptide Atlas. The idea is that if a peptide resembles a human peptide, it may have evolved to mimic a human process or to interact with human molecules.
* Compares (BLASTp) peptide predictions against peptides predicted in tick salivary glands. We want to target peptides that are in tick salivary glands because this is the location we expect to see most molecules that have evolved to influence human biology.
* Compares (BLASTp) peptide predictions against peptides with known anti-pruritic effects. We only have 5 peptides (see the [anti-pruritic peptides folder](./inputs/antipruritic_peptides)) that are known to have anti-itch activity but we compare against them.
* Clusters (mmseqs2, 80% identity) peptide predictions to determine how similar different peptides are to each other. 80% is somewhat arbitrary but is commonly in machine learning algorithms as a cut-off for shared information between proteins.
* Predicts anti-inflammatory bioactivity (AutoPeptideML) for peptide predictions. The model that does this has 70% accuracy. We expect there to be some overlap between inflammation suppression and itch suppression so we include this information.


Lastly, we ran [notebooks](./notebooks) to combine and filter the results to produce the most promising candidates for peptides that may suppress itch. 
The notebooks were run using the `envs/tidyjupyter.yml` environment.

* [20240416-explore-peptide-results.ipynb](./notebooks/20240416-explore-peptide-results.ipynb): Initial exploration of results and metadata. This notebook is messy and was more of scratch space, but I included it because it has some useful visualizations.
* [20240416-summarize-traitmapping-info-for-peptides.ipynb](./notebooks/20240416-summarize-traitmapping-info-for-peptides.ipynb): summarizes information about peptide predictions per orthogroup.
* [20240417-combine-peptide-predictions-with-metadata.ipynb](./notebooks/20240417-combine-peptide-predictions-with-metadata.ipynb): combines peptide predictions with metadata.

We initially predicted 972 peptides from the 9,603 input protein sequences.
We applied the following filters to identify a set of peptides that we think is the most promising to suppress itch:
* Filters (removes) propeptides predicted by DeepPeptide. DeepPeptide uses the [UniProt definition of a propeptide](https://www.uniprot.org/help/propep), a part of a protein that is cleaved during maturation or activation. Once cleaved, a propeptide generally has no independent biological function. This reduced the number of peptide predictions to 531.
* Filters (removes) peptides in orthogroups where no peptide had a hit to a predicted peptide from a [tick salivary gland transcriptome](https://github.com/Arcadia-Science/2024-tick-sg-peptides-tsa/). We want to target things expressed in the tick salivary gland because they are more likely to be biologically active in itch suppression. However, we don’t know if the tick salivary gland transcrpitomes we worked with are complete (many are heavily filtered) so we relaxed this filter to function at the orthogroup level. This filter reduced the number of peptides down to 289 (285 unique peptide sequences).

### Compute Specifications

* peptigate pipeline: Ran on an AWS EC2 instance type `g4dn.2xlarge` running AMI Deep Learning Base OSS Nvidia Driver GPU AMI (Ubuntu 20.04) 20240122 (AMI ID ami-07eb000b3340966b0). Note the pipeline runs many tools that use GPUs.
* [`analyze-peptigate-outputs.snakefile`](./analyze-peptigate-outputs.snakefile): Ran on an AWS EC2 instance type `g4dn.2xlarge` running AMI Deep Learning Base OSS Nvidia Driver GPU AMI (Ubuntu 20.04) 20240122 (AMI ID ami-07eb000b3340966b0). One tool in the pipeline, AutoPeptideML, uses a GPU.
* Notebooks: Ran on a MacBook Pro (2021, M1) using operating system Ventura 13.4 and with 64GB of RAM.   

## Contributing

See how we recognize [feedback and contributions to our code](https://github.com/Arcadia-Science/arcadia-software-handbook/blob/main/guides-and-standards/guide-credit-for-contributions.md).
