{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f05c0072-5b38-4776-bdc3-2a5ba8bd9d90",
   "metadata": {},
   "source": [
    "# Combine trait-mapping data with peptide predictions and summarize\n",
    "\n",
    "This notebook creates a summary TSV file that records incidence of peptide predictions within each orthogroup that is associated with the itch suppression trait.\n",
    "This information should help us select orthogroups that have:\n",
    "1. stronger peptide predictions (peptide predictions many proteins in the orthogroup)\n",
    "2. strong support for itch suppression. The strongest signal would be protein families from tick + non-tick that suppress itch. To a certain extent the coefficients should help identify these.\n",
    "\n",
    "Some thoughts from Austin included below.\n",
    "From Austin in slack (edited for clarity):\n",
    "> As for mixed orthogroups. Yes, I would absolutely expect these orthogroups to possess proteins/peptides present in both itch suppressing and non-itch suppressing species.\n",
    "> * These tests are not looking for pure presence/absence patterns (i.e. completely absent in non-suppressing and present in suppressing species) - they are looking for gross patterns.\n",
    "> * At the level of profile clusters, I took the mean counts of different event types, e.g. speciation events in the gene family (any bifurcation in the gene family tree) within itch suppressing species, and within non-itch suppressing species.\n",
    "> * I then carried out logistic regressions asking whether the mean event counts predicted itch suppression (binary, 0/1).\n",
    ">    * So long as the counts significantly differed between itch suppressing and non-itch suppressing species (and the coefficients are sufficiently large so as to be retained, depending on how you filtered), they will be included in your set of “itch suppression-associated” clusters\n",
    "> * Within profile cluster, I then conducted logistic regressions for each gene family individual, this time regressing the event counts for each species, against the response - the binary itch suppression trait.\n",
    ">    * One key bit is that this time i used a phylogenetic logistic regression that accounts for the evolutionary non-independence.\n",
    ">    * That will ultimately mean that in a hypothetical scenario where all ticks have one or more gene copy of something, but no other species do (including other non-tick itch suppressing species), this is unlikely (or less likely) to lead to a significant association. This is because although we have a bunch of the ticks in the dataset, they are effectively evolutionary pseudoreplicates, as that gene only is associated with a single evolutionary “origin” or incidence of itch suppression.\n",
    ">    * The association between that gene and itch suppression is confounded with the association between that gene and… ticks. And all other traits that are unique to them, independent of itch suppression. \n",
    "> * This is all to say… These are just statistical associations between gene family event counts in profile clusters or gene families within them and itch suppression.\n",
    ">    * Because they are associations between these different event types and not just gene presence/absence, they will include cases where the gene is typically absent in one group and present in the other, as well as cases where it’s present in both, but the “magnitude” or frequency of event counts differs.\n",
    ">\n",
    "> As final thought: I think one thing that would be particularly useful to consider in your filtering is to not only consider those gene families where peptides are present in multiple ticks, but multiple ticks as well as other non-tick itch suppressing species. To be fair, the statistical tests should already be “prioritizing” these."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "40e26562-80db-483b-8b93-1a675e9a00cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ────────────────────────────────────────────────────────────────────────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.5\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.5.0     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.3     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.2     \n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(UpSetR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "94ee01f0-f941-40cf-a952-a5986a6fcd68",
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"..\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9485cb66-4b88-4b34-895c-e7a5efcc756c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# adjust plot size rendered inline\n",
    "options(repr.plot.width = 6, repr.plot.height = 4, repr.plot.res = 300)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e1cbf7e-f6e5-4a9a-827b-d79942a2ce18",
   "metadata": {},
   "source": [
    "## Define some descriptive variables associated with trait mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e547dd3e-4e4c-4617-ad69-fbaa2548b3d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "evidence_of_itch_suppression_species <- c(\"Sarcoptes scabiei\",\n",
    "                                          \"Psoroptes ovis\",\n",
    "                                          \"Amblyomma americanum\",\n",
    "                                          \"Amblyomma sculptum\",\n",
    "                                          \"Dermacentor andersoni\",\n",
    "                                          \"Dermacentor silvarum\",\n",
    "                                          \"Dermacentor variabilis\",\n",
    "                                          \"Haemaphysalis longicornis\",\n",
    "                                          \"Hyalomma asiaticum\",\n",
    "                                          \"Ixodes persulcatus\",\n",
    "                                          \"Ixodes ricinus\",\n",
    "                                          \"Ixodes scapularis\",\n",
    "                                          \"Rhipicephalus microplus\",\n",
    "                                          \"Rhipicephalus sanguineus\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "68a5f839-c231-4100-b04c-aaa1be50d34e",
   "metadata": {},
   "outputs": [],
   "source": [
    "ticks <- c(\"Amblyomma americanum\",\n",
    "           \"Amblyomma sculptum\",\n",
    "           \"Dermacentor andersoni\",\n",
    "           \"Dermacentor silvarum\",\n",
    "           \"Dermacentor variabilis\",\n",
    "           \"Haemaphysalis longicornis\",\n",
    "           \"Hyalomma asiaticum\",\n",
    "           \"Ixodes persulcatus\",\n",
    "           \"Ixodes ricinus\",\n",
    "           \"Ixodes scapularis\",\n",
    "           \"Rhipicephalus microplus\",\n",
    "           \"Rhipicephalus sanguineus\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40fccdce-c565-44d0-ae4a-c664645de59e",
   "metadata": {},
   "source": [
    "## Read in initial data\n",
    "\n",
    "Note that for the peptigate results, we remove propeptide predictions -- DeepPeptide predict peptides and propeptides.\n",
    "The tool uses the [UniProt definition of propeptide](https://www.uniprot.org/help/propep).\n",
    "> A **Propeptide** is a part of a protein that is cleaved during maturation or activation of the protein. It is generally understood not to have an independent function.\n",
    "A **Peptide** is proteolytically cleaved and has a well-defined biological activity.\n",
    "\n",
    "Given this, we remove propeptide predictions as DeepPeptide would predict that these are not biologically active after being cleaved from their precursor protein."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f6e11206-bd69-4f71-8cf1-c40dcfe2310c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# tot peptigate predictions & remove propeptide predictions\n",
    "peptigate_predictions <- read_tsv(\"outputs/ToT_20240626/predictions/peptide_predictions.tsv\", show_col_types = F) %>%\n",
    "  rename_with(.cols = everything(), function(x){paste0(\"peptigate_\", x)}) %>%\n",
    "  mutate(peptigate_peptide_class = ifelse(is.na(peptigate_peptide_class), \"sORF\", peptigate_peptide_class)) %>%\n",
    "  filter(peptigate_peptide_class != \"Propeptide\") %>%\n",
    "  mutate(traitmapping_locus_tag = gsub(\"_start.*\", \"\", peptigate_peptide_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "251c65f5-f0f1-4ccd-b0d6-43101e69ced1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>peptigate_peptide_id</th><th scope=col>peptigate_start</th><th scope=col>peptigate_end</th><th scope=col>peptigate_peptide_type</th><th scope=col>peptigate_peptide_class</th><th scope=col>peptigate_prediction_tool</th><th scope=col>peptigate_nlpprecursor_class_score</th><th scope=col>peptigate_nlpprecursor_cleavage_score</th><th scope=col>peptigate_X3.x</th><th scope=col>peptigate_protein_sequence</th><th scope=col>peptigate_X3.y</th><th scope=col>traitmapping_locus_tag</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Rhipicephalus-sanguineus_XP-037517499.1</td><td>NA</td><td>NA</td><td>sORF</td><td>sORF</td><td>less_than_100aa</td><td>NA</td><td>NA</td><td>NA</td><td>MKMIIFTITLLIISDYVWPCDATGSTICSRAFMVIRKSFGCPFGSNECSKYCKEKKAQKGGYCTGRFKENCVCFRN                        </td><td>NA</td><td>Rhipicephalus-sanguineus_XP-037517499.1</td></tr>\n",
       "\t<tr><td>Rhipicephalus-sanguineus_XP-037521633.1</td><td>NA</td><td>NA</td><td>sORF</td><td>sORF</td><td>less_than_100aa</td><td>NA</td><td>NA</td><td>NA</td><td>MKLLTIFLIFGLVSGILATASLNENKSEVAHLRVRRWGFGCPFNQRACHRHCRSIRRRAGYCGGRFKLTCTCVRR                         </td><td>NA</td><td>Rhipicephalus-sanguineus_XP-037521633.1</td></tr>\n",
       "\t<tr><td>Ricinoides-atewa_comp77536-c0-seq1.p1  </td><td>NA</td><td>NA</td><td>sORF</td><td>sORF</td><td>less_than_100aa</td><td>NA</td><td>NA</td><td>NA</td><td>RAWENLAAEYNSRPGVHRRTGTQLRKLWVNLKTRNKKEGKEMALDRPTIVREKPERPTSISPTTMKVIQLVPSLMRSFEGGDSDLEEDLQSTVHIMLWPT</td><td>NA</td><td>Ricinoides-atewa_comp77536-c0-seq1.p1  </td></tr>\n",
       "\t<tr><td>Ricinoides-atewa_comp81574-c0-seq1.p1  </td><td>NA</td><td>NA</td><td>sORF</td><td>sORF</td><td>less_than_100aa</td><td>NA</td><td>NA</td><td>NA</td><td>MIRERAKMIFNTEIDEESKRKETFVASSGWLQKFMERNSLSLRRKTTVAQHDPDKLIDKVVSFVMWISQLRMTKSIPECNIIGMDETSVWFDMPGETTV </td><td>NA</td><td>Ricinoides-atewa_comp81574-c0-seq1.p1  </td></tr>\n",
       "\t<tr><td>Tyrophagus-putrescentiae_KAH9391113.1  </td><td>NA</td><td>NA</td><td>sORF</td><td>sORF</td><td>less_than_100aa</td><td>NA</td><td>NA</td><td>NA</td><td>MALLNVNFVWSPVTVFVAAGKTFGCPLTSSCRKHCQMNKFKGGQCEGNLRLLMNPICTNIAPN                                     </td><td>NA</td><td>Tyrophagus-putrescentiae_KAH9391113.1  </td></tr>\n",
       "\t<tr><td>Tyrophagus-putrescentiae_KAH9391114.1  </td><td>NA</td><td>NA</td><td>sORF</td><td>sORF</td><td>less_than_100aa</td><td>NA</td><td>NA</td><td>NA</td><td>MKLFGQVFSLLLLLTIGLVMMSTSVSADYGCPITSKCKQHCLENKFKSGSCEGTLKLTCHCVG                                     </td><td>NA</td><td>Tyrophagus-putrescentiae_KAH9391114.1  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 12\n",
       "\\begin{tabular}{llllllllllll}\n",
       " peptigate\\_peptide\\_id & peptigate\\_start & peptigate\\_end & peptigate\\_peptide\\_type & peptigate\\_peptide\\_class & peptigate\\_prediction\\_tool & peptigate\\_nlpprecursor\\_class\\_score & peptigate\\_nlpprecursor\\_cleavage\\_score & peptigate\\_X3.x & peptigate\\_protein\\_sequence & peptigate\\_X3.y & traitmapping\\_locus\\_tag\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <lgl> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t Rhipicephalus-sanguineus\\_XP-037517499.1 & NA & NA & sORF & sORF & less\\_than\\_100aa & NA & NA & NA & MKMIIFTITLLIISDYVWPCDATGSTICSRAFMVIRKSFGCPFGSNECSKYCKEKKAQKGGYCTGRFKENCVCFRN                         & NA & Rhipicephalus-sanguineus\\_XP-037517499.1\\\\\n",
       "\t Rhipicephalus-sanguineus\\_XP-037521633.1 & NA & NA & sORF & sORF & less\\_than\\_100aa & NA & NA & NA & MKLLTIFLIFGLVSGILATASLNENKSEVAHLRVRRWGFGCPFNQRACHRHCRSIRRRAGYCGGRFKLTCTCVRR                          & NA & Rhipicephalus-sanguineus\\_XP-037521633.1\\\\\n",
       "\t Ricinoides-atewa\\_comp77536-c0-seq1.p1   & NA & NA & sORF & sORF & less\\_than\\_100aa & NA & NA & NA & RAWENLAAEYNSRPGVHRRTGTQLRKLWVNLKTRNKKEGKEMALDRPTIVREKPERPTSISPTTMKVIQLVPSLMRSFEGGDSDLEEDLQSTVHIMLWPT & NA & Ricinoides-atewa\\_comp77536-c0-seq1.p1  \\\\\n",
       "\t Ricinoides-atewa\\_comp81574-c0-seq1.p1   & NA & NA & sORF & sORF & less\\_than\\_100aa & NA & NA & NA & MIRERAKMIFNTEIDEESKRKETFVASSGWLQKFMERNSLSLRRKTTVAQHDPDKLIDKVVSFVMWISQLRMTKSIPECNIIGMDETSVWFDMPGETTV  & NA & Ricinoides-atewa\\_comp81574-c0-seq1.p1  \\\\\n",
       "\t Tyrophagus-putrescentiae\\_KAH9391113.1   & NA & NA & sORF & sORF & less\\_than\\_100aa & NA & NA & NA & MALLNVNFVWSPVTVFVAAGKTFGCPLTSSCRKHCQMNKFKGGQCEGNLRLLMNPICTNIAPN                                      & NA & Tyrophagus-putrescentiae\\_KAH9391113.1  \\\\\n",
       "\t Tyrophagus-putrescentiae\\_KAH9391114.1   & NA & NA & sORF & sORF & less\\_than\\_100aa & NA & NA & NA & MKLFGQVFSLLLLLTIGLVMMSTSVSADYGCPITSKCKQHCLENKFKSGSCEGTLKLTCHCVG                                      & NA & Tyrophagus-putrescentiae\\_KAH9391114.1  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 12\n",
       "\n",
       "| peptigate_peptide_id &lt;chr&gt; | peptigate_start &lt;dbl&gt; | peptigate_end &lt;dbl&gt; | peptigate_peptide_type &lt;chr&gt; | peptigate_peptide_class &lt;chr&gt; | peptigate_prediction_tool &lt;chr&gt; | peptigate_nlpprecursor_class_score &lt;dbl&gt; | peptigate_nlpprecursor_cleavage_score &lt;dbl&gt; | peptigate_X3.x &lt;lgl&gt; | peptigate_protein_sequence &lt;chr&gt; | peptigate_X3.y &lt;lgl&gt; | traitmapping_locus_tag &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Rhipicephalus-sanguineus_XP-037517499.1 | NA | NA | sORF | sORF | less_than_100aa | NA | NA | NA | MKMIIFTITLLIISDYVWPCDATGSTICSRAFMVIRKSFGCPFGSNECSKYCKEKKAQKGGYCTGRFKENCVCFRN                         | NA | Rhipicephalus-sanguineus_XP-037517499.1 |\n",
       "| Rhipicephalus-sanguineus_XP-037521633.1 | NA | NA | sORF | sORF | less_than_100aa | NA | NA | NA | MKLLTIFLIFGLVSGILATASLNENKSEVAHLRVRRWGFGCPFNQRACHRHCRSIRRRAGYCGGRFKLTCTCVRR                          | NA | Rhipicephalus-sanguineus_XP-037521633.1 |\n",
       "| Ricinoides-atewa_comp77536-c0-seq1.p1   | NA | NA | sORF | sORF | less_than_100aa | NA | NA | NA | RAWENLAAEYNSRPGVHRRTGTQLRKLWVNLKTRNKKEGKEMALDRPTIVREKPERPTSISPTTMKVIQLVPSLMRSFEGGDSDLEEDLQSTVHIMLWPT | NA | Ricinoides-atewa_comp77536-c0-seq1.p1   |\n",
       "| Ricinoides-atewa_comp81574-c0-seq1.p1   | NA | NA | sORF | sORF | less_than_100aa | NA | NA | NA | MIRERAKMIFNTEIDEESKRKETFVASSGWLQKFMERNSLSLRRKTTVAQHDPDKLIDKVVSFVMWISQLRMTKSIPECNIIGMDETSVWFDMPGETTV  | NA | Ricinoides-atewa_comp81574-c0-seq1.p1   |\n",
       "| Tyrophagus-putrescentiae_KAH9391113.1   | NA | NA | sORF | sORF | less_than_100aa | NA | NA | NA | MALLNVNFVWSPVTVFVAAGKTFGCPLTSSCRKHCQMNKFKGGQCEGNLRLLMNPICTNIAPN                                      | NA | Tyrophagus-putrescentiae_KAH9391113.1   |\n",
       "| Tyrophagus-putrescentiae_KAH9391114.1   | NA | NA | sORF | sORF | less_than_100aa | NA | NA | NA | MKLFGQVFSLLLLLTIGLVMMSTSVSADYGCPITSKCKQHCLENKFKSGSCEGTLKLTCHCVG                                      | NA | Tyrophagus-putrescentiae_KAH9391114.1   |\n",
       "\n"
      ],
      "text/plain": [
       "  peptigate_peptide_id                    peptigate_start peptigate_end\n",
       "1 Rhipicephalus-sanguineus_XP-037517499.1 NA              NA           \n",
       "2 Rhipicephalus-sanguineus_XP-037521633.1 NA              NA           \n",
       "3 Ricinoides-atewa_comp77536-c0-seq1.p1   NA              NA           \n",
       "4 Ricinoides-atewa_comp81574-c0-seq1.p1   NA              NA           \n",
       "5 Tyrophagus-putrescentiae_KAH9391113.1   NA              NA           \n",
       "6 Tyrophagus-putrescentiae_KAH9391114.1   NA              NA           \n",
       "  peptigate_peptide_type peptigate_peptide_class peptigate_prediction_tool\n",
       "1 sORF                   sORF                    less_than_100aa          \n",
       "2 sORF                   sORF                    less_than_100aa          \n",
       "3 sORF                   sORF                    less_than_100aa          \n",
       "4 sORF                   sORF                    less_than_100aa          \n",
       "5 sORF                   sORF                    less_than_100aa          \n",
       "6 sORF                   sORF                    less_than_100aa          \n",
       "  peptigate_nlpprecursor_class_score peptigate_nlpprecursor_cleavage_score\n",
       "1 NA                                 NA                                   \n",
       "2 NA                                 NA                                   \n",
       "3 NA                                 NA                                   \n",
       "4 NA                                 NA                                   \n",
       "5 NA                                 NA                                   \n",
       "6 NA                                 NA                                   \n",
       "  peptigate_X3.x\n",
       "1 NA            \n",
       "2 NA            \n",
       "3 NA            \n",
       "4 NA            \n",
       "5 NA            \n",
       "6 NA            \n",
       "  peptigate_protein_sequence                                                                          \n",
       "1 MKMIIFTITLLIISDYVWPCDATGSTICSRAFMVIRKSFGCPFGSNECSKYCKEKKAQKGGYCTGRFKENCVCFRN                        \n",
       "2 MKLLTIFLIFGLVSGILATASLNENKSEVAHLRVRRWGFGCPFNQRACHRHCRSIRRRAGYCGGRFKLTCTCVRR                         \n",
       "3 RAWENLAAEYNSRPGVHRRTGTQLRKLWVNLKTRNKKEGKEMALDRPTIVREKPERPTSISPTTMKVIQLVPSLMRSFEGGDSDLEEDLQSTVHIMLWPT\n",
       "4 MIRERAKMIFNTEIDEESKRKETFVASSGWLQKFMERNSLSLRRKTTVAQHDPDKLIDKVVSFVMWISQLRMTKSIPECNIIGMDETSVWFDMPGETTV \n",
       "5 MALLNVNFVWSPVTVFVAAGKTFGCPLTSSCRKHCQMNKFKGGQCEGNLRLLMNPICTNIAPN                                     \n",
       "6 MKLFGQVFSLLLLLTIGLVMMSTSVSADYGCPITSKCKQHCLENKFKSGSCEGTLKLTCHCVG                                     \n",
       "  peptigate_X3.y traitmapping_locus_tag                 \n",
       "1 NA             Rhipicephalus-sanguineus_XP-037517499.1\n",
       "2 NA             Rhipicephalus-sanguineus_XP-037521633.1\n",
       "3 NA             Ricinoides-atewa_comp77536-c0-seq1.p1  \n",
       "4 NA             Ricinoides-atewa_comp81574-c0-seq1.p1  \n",
       "5 NA             Tyrophagus-putrescentiae_KAH9391113.1  \n",
       "6 NA             Tyrophagus-putrescentiae_KAH9391114.1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail(peptigate_predictions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0cffc5d2-eaf1-4051-998c-28cfd29f44f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# transcriptome shotgun assembly (TSA) salivary gland (sg) transcriptome peptide prediction BLAST hits\n",
    "tsa_sg_blastp <- read_tsv(\"outputs/analysis/compare_tsa_sg/tsa_sg_peptides_blastp_matches.tsv\", show_col_types = FALSE) %>%\n",
    "  # select only one blast hit for each query peptide\n",
    "  group_by(qseqid) %>%\n",
    "  slice_max(bitscore) %>%\n",
    "  slice_min(evalue) %>%\n",
    "  slice_head(n = 1) %>%\n",
    "  ungroup() %>%\n",
    "  filter(qseqid %in% peptigate_predictions$peptigate_peptide_id) %>% # filter out hits to propeptides\n",
    "  rename_with(.cols = everything(), function(x){paste0(\"sgpeptide_blast_\", x)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e364db31-b4d8-4585-b537-fbba7d50027e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# metadata from trait mapping ToT data for association with itch-suppression\n",
    "trait_mapping_metadata <- read_tsv(\"inputs/2024-06-26-top-positive-significant-clusters-orthogroups-annotations.tsv.gz\", show_col_types = F) %>%\n",
    "  mutate(species = gsub(\"-\", \" \", species)) %>%\n",
    "  rename_with(.cols = everything(), function(x){paste0(\"traitmapping_\", x)})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "791b9ee3-6c2c-44a3-8dba-d3af470f4bf0",
   "metadata": {},
   "source": [
    "## Use the trait mapping metadata as well as peptigate results and analysis of those results to understand the breakdown of the different orthogroups\n",
    "\n",
    "* number of proteins in orthogroup\n",
    "* number of proteins in orthogroup with evidence of expression in salivary gland transcriptomes from the TSA (and percent)\n",
    "* number of proteins in orthogroup with peptide prediction (and percent)\n",
    "* number of proteins in orthogroup from ticks (order Ixodida) (and percent)\n",
    "* number of proteins in orthogroup from chelicerates that suppress itch (and percent)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4f16a542-d33e-4b25-ba23-9cef74efa9e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'traitmapping_cluster', 'traitmapping_orthogroup', 'traitmapping_coefficient', 'traitmapping_signif_level'.\n",
      "You can override using the `.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "# number of proteins in orthogroup with peptide prediction (and percent)\n",
    "orthogroup_peptide_summary <- trait_mapping_metadata %>%\n",
    "  # join to peptide predictions \n",
    "  left_join(peptigate_predictions, by = \"traitmapping_locus_tag\") %>%\n",
    "  # join to peptide prediction hits in the tick salivary gland transcriptomes\n",
    "  left_join(tsa_sg_blastp, by = c(\"peptigate_peptide_id\" = \"sgpeptide_blast_qseqid\")) %>%\n",
    "  mutate(peptigate_peptide_prediction = ifelse(is.na(peptigate_peptide_class), \"no peptide prediction\", \"peptide prediction\"),\n",
    "         species_suppresses_itch = ifelse(traitmapping_species %in% evidence_of_itch_suppression_species, \"evidence suppresses itch\", \"no evidence suppresses itch\"),\n",
    "         species_is_a_tick = ifelse(traitmapping_species %in% ticks, \"tick\", \"not tick\")) %>%\n",
    "  group_by(traitmapping_cluster, traitmapping_orthogroup, traitmapping_coefficient, traitmapping_signif_level, traitmapping_signif_fdr) %>%\n",
    "  summarize(num_proteins_in_orthogroup = n(),\n",
    "            # number of proteins in orthogroup with peptide prediction (and fraction)\n",
    "            num_predicted_peptides = sum(peptigate_peptide_prediction == \"peptide prediction\"),\n",
    "            fraction_of_orthogroup_with_predicted_peptide = num_predicted_peptides / num_proteins_in_orthogroup,\n",
    "            # number of protein in orthogroup with signal peptide (and fraction)\n",
    "            num_predicted_signal_peptides = sum(traitmapping_deepsig_feature == \"Signal peptide\"),\n",
    "            fraction_of_orthogroup_with_signal_peptide = num_predicted_signal_peptides / num_proteins_in_orthogroup,\n",
    "            # number of predicted peptides with predicted signal peptides in orthogroup (and fraction)\n",
    "            num_predicted_peptides_with_signal_peptide = sum(traitmapping_deepsig_feature == \"Signal peptide\" & peptigate_peptide_prediction == \"peptide prediction\"),\n",
    "            fraction_of_orthogroup_with_predicted_peptide_and_signal_peptide = num_predicted_peptides_with_signal_peptide / num_proteins_in_orthogroup,\n",
    "            # number of proteins in orthogroup from ticks (order Ixodida) (and fraction), and how many had peptide predictions\n",
    "            num_tick_proteins_in_orthogroup = sum(species_is_a_tick == \"tick\"),\n",
    "            fraction_of_orthogroup_tick_proteins = num_tick_proteins_in_orthogroup / num_proteins_in_orthogroup,\n",
    "            num_predicted_peptides_from_tick = sum(peptigate_peptide_prediction == \"peptide prediction\" & species_is_a_tick == \"tick\"),\n",
    "            fraction_of_orthogroup_with_predicted_tick_peptides = num_predicted_peptides_from_tick / num_proteins_in_orthogroup,\n",
    "            # number of proteins in orthogroup from chelicerates that suppress itch (and percent)\n",
    "            # note itchsuppsp stands for \"itch suppression species\"\n",
    "            num_itchsuppsp_proteins_in_orthogroup = sum(species_suppresses_itch == \"evidence suppresses itch\"),\n",
    "            fraction_of_orthogroup_itchsuppsp_proteins = num_itchsuppsp_proteins_in_orthogroup / num_proteins_in_orthogroup,\n",
    "            num_predicted_peptides_from_itchsuppsp = sum(peptigate_peptide_prediction == \"peptide prediction\" & species_suppresses_itch == \"evidence suppresses itch\"),\n",
    "            fraction_of_orthogroup_with_predicted_itchsuppsp_peptides = num_predicted_peptides_from_itchsuppsp / num_proteins_in_orthogroup,\n",
    "            # calculate whether there is tick & non-tick evidence of itch suppression\n",
    "            type_of_itch_suppression_evidence = ifelse(num_predicted_peptides_from_itchsuppsp > num_predicted_peptides_from_tick, \"chelicerate support\", \n",
    "                                                       ifelse(num_predicted_peptides_from_tick > 0, \"tick support\", \"no support\")),\n",
    "            # number of proteins (peptides) in orthogroup with evidence of expression in salivary gland transcriptomes from the TSA (and percent)\n",
    "            num_predicted_peptides_with_sg_blast_hit = sum(!is.na(sgpeptide_blast_bitscore))\n",
    "           ) %>%\n",
    "  arrange(type_of_itch_suppression_evidence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "750c9bf4-58f6-417a-81ee-3aed5751556a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 13 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>traitmapping_cluster</th><th scope=col>traitmapping_orthogroup</th><th scope=col>traitmapping_coefficient</th><th scope=col>traitmapping_signif_level</th><th scope=col>traitmapping_signif_fdr</th><th scope=col>num_proteins_in_orthogroup</th><th scope=col>num_predicted_peptides</th><th scope=col>fraction_of_orthogroup_with_predicted_peptide</th><th scope=col>num_predicted_signal_peptides</th><th scope=col>fraction_of_orthogroup_with_signal_peptide</th><th scope=col>⋯</th><th scope=col>num_tick_proteins_in_orthogroup</th><th scope=col>fraction_of_orthogroup_tick_proteins</th><th scope=col>num_predicted_peptides_from_tick</th><th scope=col>fraction_of_orthogroup_with_predicted_tick_peptides</th><th scope=col>num_itchsuppsp_proteins_in_orthogroup</th><th scope=col>fraction_of_orthogroup_itchsuppsp_proteins</th><th scope=col>num_predicted_peptides_from_itchsuppsp</th><th scope=col>fraction_of_orthogroup_with_predicted_itchsuppsp_peptides</th><th scope=col>type_of_itch_suppression_evidence</th><th scope=col>num_predicted_peptides_with_sg_blast_hit</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cluster_40</td><td>OG0011284</td><td>2.24534051</td><td>Yes</td><td>No</td><td>  9</td><td> 1</td><td>0.11111111</td><td> 0</td><td>0.00000000</td><td>⋯</td><td>  8</td><td>0.8888889</td><td> 1</td><td>0.11111111</td><td>  8</td><td>0.8888889</td><td> 1</td><td>0.11111111</td><td>tick support       </td><td> 1</td></tr>\n",
       "\t<tr><td>cluster_40</td><td>OG0008888</td><td>1.69892817</td><td>Yes</td><td>No</td><td> 16</td><td> 1</td><td>0.06250000</td><td> 7</td><td>0.43750000</td><td>⋯</td><td> 16</td><td>1.0000000</td><td> 1</td><td>0.06250000</td><td> 16</td><td>1.0000000</td><td> 1</td><td>0.06250000</td><td>tick support       </td><td> 1</td></tr>\n",
       "\t<tr><td>cluster_40</td><td>OG0001774</td><td>1.05408448</td><td>Yes</td><td>No</td><td> 66</td><td>45</td><td>0.68181818</td><td>43</td><td>0.65151515</td><td>⋯</td><td> 51</td><td>0.7727273</td><td>38</td><td>0.57575758</td><td> 52</td><td>0.7878788</td><td>39</td><td>0.59090909</td><td>chelicerate support</td><td>34</td></tr>\n",
       "\t<tr><td>cluster_33</td><td>OG0008102</td><td>0.89907992</td><td>Yes</td><td>No</td><td> 24</td><td>18</td><td>0.75000000</td><td>17</td><td>0.70833333</td><td>⋯</td><td> 24</td><td>1.0000000</td><td>18</td><td>0.75000000</td><td> 24</td><td>1.0000000</td><td>18</td><td>0.75000000</td><td>tick support       </td><td>13</td></tr>\n",
       "\t<tr><td>cluster_27</td><td>OG0002194</td><td>0.87213413</td><td>Yes</td><td>No</td><td> 56</td><td> 2</td><td>0.03571429</td><td> 1</td><td>0.01785714</td><td>⋯</td><td> 45</td><td>0.8035714</td><td> 2</td><td>0.03571429</td><td> 45</td><td>0.8035714</td><td> 2</td><td>0.03571429</td><td>tick support       </td><td> 1</td></tr>\n",
       "\t<tr><td>cluster_27</td><td>OG0000189</td><td>0.50428119</td><td>Yes</td><td>No</td><td>240</td><td>21</td><td>0.08750000</td><td>22</td><td>0.09166667</td><td>⋯</td><td>193</td><td>0.8041667</td><td>20</td><td>0.08333333</td><td>195</td><td>0.8125000</td><td>20</td><td>0.08333333</td><td>tick support       </td><td>15</td></tr>\n",
       "\t<tr><td>cluster_46</td><td>OG0000746</td><td>0.42872398</td><td>Yes</td><td>No</td><td>102</td><td> 5</td><td>0.04901961</td><td> 2</td><td>0.01960784</td><td>⋯</td><td> 71</td><td>0.6960784</td><td> 4</td><td>0.03921569</td><td> 72</td><td>0.7058824</td><td> 4</td><td>0.03921569</td><td>tick support       </td><td> 2</td></tr>\n",
       "\t<tr><td>cluster_27</td><td>OG0000194</td><td>0.35313516</td><td>Yes</td><td>No</td><td>237</td><td>23</td><td>0.09704641</td><td>NA</td><td>        NA</td><td>⋯</td><td>203</td><td>0.8565401</td><td>21</td><td>0.08860759</td><td>203</td><td>0.8565401</td><td>21</td><td>0.08860759</td><td>tick support       </td><td>10</td></tr>\n",
       "\t<tr><td>cluster_33</td><td>OG0000880</td><td>0.31400766</td><td>Yes</td><td>No</td><td>109</td><td>84</td><td>0.77064220</td><td>57</td><td>0.52293578</td><td>⋯</td><td>107</td><td>0.9816514</td><td>82</td><td>0.75229358</td><td>107</td><td>0.9816514</td><td>82</td><td>0.75229358</td><td>tick support       </td><td>30</td></tr>\n",
       "\t<tr><td>cluster_46</td><td>OG0001663</td><td>0.29542423</td><td>Yes</td><td>No</td><td> 64</td><td> 6</td><td>0.09375000</td><td> 2</td><td>0.03125000</td><td>⋯</td><td> 64</td><td>1.0000000</td><td> 6</td><td>0.09375000</td><td> 64</td><td>1.0000000</td><td> 6</td><td>0.09375000</td><td>tick support       </td><td> 4</td></tr>\n",
       "\t<tr><td>cluster_40</td><td>OG0000385</td><td>0.25070199</td><td>Yes</td><td>No</td><td>154</td><td> 9</td><td>0.05844156</td><td> 4</td><td>0.02597403</td><td>⋯</td><td>146</td><td>0.9480519</td><td> 8</td><td>0.05194805</td><td>146</td><td>0.9480519</td><td> 8</td><td>0.05194805</td><td>tick support       </td><td> 2</td></tr>\n",
       "\t<tr><td>cluster_40</td><td>OG0000143</td><td>0.15338398</td><td>Yes</td><td>No</td><td>282</td><td>11</td><td>0.03900709</td><td>11</td><td>0.03900709</td><td>⋯</td><td>232</td><td>0.8226950</td><td>10</td><td>0.03546099</td><td>232</td><td>0.8226950</td><td>10</td><td>0.03546099</td><td>tick support       </td><td> 5</td></tr>\n",
       "\t<tr><td>cluster_27</td><td>OG0000079</td><td>0.08290626</td><td>Yes</td><td>No</td><td>394</td><td>55</td><td>0.13959391</td><td>NA</td><td>        NA</td><td>⋯</td><td>293</td><td>0.7436548</td><td>50</td><td>0.12690355</td><td>293</td><td>0.7436548</td><td>50</td><td>0.12690355</td><td>tick support       </td><td>28</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 13 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " traitmapping\\_cluster & traitmapping\\_orthogroup & traitmapping\\_coefficient & traitmapping\\_signif\\_level & traitmapping\\_signif\\_fdr & num\\_proteins\\_in\\_orthogroup & num\\_predicted\\_peptides & fraction\\_of\\_orthogroup\\_with\\_predicted\\_peptide & num\\_predicted\\_signal\\_peptides & fraction\\_of\\_orthogroup\\_with\\_signal\\_peptide & ⋯ & num\\_tick\\_proteins\\_in\\_orthogroup & fraction\\_of\\_orthogroup\\_tick\\_proteins & num\\_predicted\\_peptides\\_from\\_tick & fraction\\_of\\_orthogroup\\_with\\_predicted\\_tick\\_peptides & num\\_itchsuppsp\\_proteins\\_in\\_orthogroup & fraction\\_of\\_orthogroup\\_itchsuppsp\\_proteins & num\\_predicted\\_peptides\\_from\\_itchsuppsp & fraction\\_of\\_orthogroup\\_with\\_predicted\\_itchsuppsp\\_peptides & type\\_of\\_itch\\_suppression\\_evidence & num\\_predicted\\_peptides\\_with\\_sg\\_blast\\_hit\\\\\n",
       " <chr> & <chr> & <dbl> & <chr> & <chr> & <int> & <int> & <dbl> & <int> & <dbl> & ⋯ & <int> & <dbl> & <int> & <dbl> & <int> & <dbl> & <int> & <dbl> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t cluster\\_40 & OG0011284 & 2.24534051 & Yes & No &   9 &  1 & 0.11111111 &  0 & 0.00000000 & ⋯ &   8 & 0.8888889 &  1 & 0.11111111 &   8 & 0.8888889 &  1 & 0.11111111 & tick support        &  1\\\\\n",
       "\t cluster\\_40 & OG0008888 & 1.69892817 & Yes & No &  16 &  1 & 0.06250000 &  7 & 0.43750000 & ⋯ &  16 & 1.0000000 &  1 & 0.06250000 &  16 & 1.0000000 &  1 & 0.06250000 & tick support        &  1\\\\\n",
       "\t cluster\\_40 & OG0001774 & 1.05408448 & Yes & No &  66 & 45 & 0.68181818 & 43 & 0.65151515 & ⋯ &  51 & 0.7727273 & 38 & 0.57575758 &  52 & 0.7878788 & 39 & 0.59090909 & chelicerate support & 34\\\\\n",
       "\t cluster\\_33 & OG0008102 & 0.89907992 & Yes & No &  24 & 18 & 0.75000000 & 17 & 0.70833333 & ⋯ &  24 & 1.0000000 & 18 & 0.75000000 &  24 & 1.0000000 & 18 & 0.75000000 & tick support        & 13\\\\\n",
       "\t cluster\\_27 & OG0002194 & 0.87213413 & Yes & No &  56 &  2 & 0.03571429 &  1 & 0.01785714 & ⋯ &  45 & 0.8035714 &  2 & 0.03571429 &  45 & 0.8035714 &  2 & 0.03571429 & tick support        &  1\\\\\n",
       "\t cluster\\_27 & OG0000189 & 0.50428119 & Yes & No & 240 & 21 & 0.08750000 & 22 & 0.09166667 & ⋯ & 193 & 0.8041667 & 20 & 0.08333333 & 195 & 0.8125000 & 20 & 0.08333333 & tick support        & 15\\\\\n",
       "\t cluster\\_46 & OG0000746 & 0.42872398 & Yes & No & 102 &  5 & 0.04901961 &  2 & 0.01960784 & ⋯ &  71 & 0.6960784 &  4 & 0.03921569 &  72 & 0.7058824 &  4 & 0.03921569 & tick support        &  2\\\\\n",
       "\t cluster\\_27 & OG0000194 & 0.35313516 & Yes & No & 237 & 23 & 0.09704641 & NA &         NA & ⋯ & 203 & 0.8565401 & 21 & 0.08860759 & 203 & 0.8565401 & 21 & 0.08860759 & tick support        & 10\\\\\n",
       "\t cluster\\_33 & OG0000880 & 0.31400766 & Yes & No & 109 & 84 & 0.77064220 & 57 & 0.52293578 & ⋯ & 107 & 0.9816514 & 82 & 0.75229358 & 107 & 0.9816514 & 82 & 0.75229358 & tick support        & 30\\\\\n",
       "\t cluster\\_46 & OG0001663 & 0.29542423 & Yes & No &  64 &  6 & 0.09375000 &  2 & 0.03125000 & ⋯ &  64 & 1.0000000 &  6 & 0.09375000 &  64 & 1.0000000 &  6 & 0.09375000 & tick support        &  4\\\\\n",
       "\t cluster\\_40 & OG0000385 & 0.25070199 & Yes & No & 154 &  9 & 0.05844156 &  4 & 0.02597403 & ⋯ & 146 & 0.9480519 &  8 & 0.05194805 & 146 & 0.9480519 &  8 & 0.05194805 & tick support        &  2\\\\\n",
       "\t cluster\\_40 & OG0000143 & 0.15338398 & Yes & No & 282 & 11 & 0.03900709 & 11 & 0.03900709 & ⋯ & 232 & 0.8226950 & 10 & 0.03546099 & 232 & 0.8226950 & 10 & 0.03546099 & tick support        &  5\\\\\n",
       "\t cluster\\_27 & OG0000079 & 0.08290626 & Yes & No & 394 & 55 & 0.13959391 & NA &         NA & ⋯ & 293 & 0.7436548 & 50 & 0.12690355 & 293 & 0.7436548 & 50 & 0.12690355 & tick support        & 28\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 13 × 22\n",
       "\n",
       "| traitmapping_cluster &lt;chr&gt; | traitmapping_orthogroup &lt;chr&gt; | traitmapping_coefficient &lt;dbl&gt; | traitmapping_signif_level &lt;chr&gt; | traitmapping_signif_fdr &lt;chr&gt; | num_proteins_in_orthogroup &lt;int&gt; | num_predicted_peptides &lt;int&gt; | fraction_of_orthogroup_with_predicted_peptide &lt;dbl&gt; | num_predicted_signal_peptides &lt;int&gt; | fraction_of_orthogroup_with_signal_peptide &lt;dbl&gt; | ⋯ ⋯ | num_tick_proteins_in_orthogroup &lt;int&gt; | fraction_of_orthogroup_tick_proteins &lt;dbl&gt; | num_predicted_peptides_from_tick &lt;int&gt; | fraction_of_orthogroup_with_predicted_tick_peptides &lt;dbl&gt; | num_itchsuppsp_proteins_in_orthogroup &lt;int&gt; | fraction_of_orthogroup_itchsuppsp_proteins &lt;dbl&gt; | num_predicted_peptides_from_itchsuppsp &lt;int&gt; | fraction_of_orthogroup_with_predicted_itchsuppsp_peptides &lt;dbl&gt; | type_of_itch_suppression_evidence &lt;chr&gt; | num_predicted_peptides_with_sg_blast_hit &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cluster_40 | OG0011284 | 2.24534051 | Yes | No |   9 |  1 | 0.11111111 |  0 | 0.00000000 | ⋯ |   8 | 0.8888889 |  1 | 0.11111111 |   8 | 0.8888889 |  1 | 0.11111111 | tick support        |  1 |\n",
       "| cluster_40 | OG0008888 | 1.69892817 | Yes | No |  16 |  1 | 0.06250000 |  7 | 0.43750000 | ⋯ |  16 | 1.0000000 |  1 | 0.06250000 |  16 | 1.0000000 |  1 | 0.06250000 | tick support        |  1 |\n",
       "| cluster_40 | OG0001774 | 1.05408448 | Yes | No |  66 | 45 | 0.68181818 | 43 | 0.65151515 | ⋯ |  51 | 0.7727273 | 38 | 0.57575758 |  52 | 0.7878788 | 39 | 0.59090909 | chelicerate support | 34 |\n",
       "| cluster_33 | OG0008102 | 0.89907992 | Yes | No |  24 | 18 | 0.75000000 | 17 | 0.70833333 | ⋯ |  24 | 1.0000000 | 18 | 0.75000000 |  24 | 1.0000000 | 18 | 0.75000000 | tick support        | 13 |\n",
       "| cluster_27 | OG0002194 | 0.87213413 | Yes | No |  56 |  2 | 0.03571429 |  1 | 0.01785714 | ⋯ |  45 | 0.8035714 |  2 | 0.03571429 |  45 | 0.8035714 |  2 | 0.03571429 | tick support        |  1 |\n",
       "| cluster_27 | OG0000189 | 0.50428119 | Yes | No | 240 | 21 | 0.08750000 | 22 | 0.09166667 | ⋯ | 193 | 0.8041667 | 20 | 0.08333333 | 195 | 0.8125000 | 20 | 0.08333333 | tick support        | 15 |\n",
       "| cluster_46 | OG0000746 | 0.42872398 | Yes | No | 102 |  5 | 0.04901961 |  2 | 0.01960784 | ⋯ |  71 | 0.6960784 |  4 | 0.03921569 |  72 | 0.7058824 |  4 | 0.03921569 | tick support        |  2 |\n",
       "| cluster_27 | OG0000194 | 0.35313516 | Yes | No | 237 | 23 | 0.09704641 | NA |         NA | ⋯ | 203 | 0.8565401 | 21 | 0.08860759 | 203 | 0.8565401 | 21 | 0.08860759 | tick support        | 10 |\n",
       "| cluster_33 | OG0000880 | 0.31400766 | Yes | No | 109 | 84 | 0.77064220 | 57 | 0.52293578 | ⋯ | 107 | 0.9816514 | 82 | 0.75229358 | 107 | 0.9816514 | 82 | 0.75229358 | tick support        | 30 |\n",
       "| cluster_46 | OG0001663 | 0.29542423 | Yes | No |  64 |  6 | 0.09375000 |  2 | 0.03125000 | ⋯ |  64 | 1.0000000 |  6 | 0.09375000 |  64 | 1.0000000 |  6 | 0.09375000 | tick support        |  4 |\n",
       "| cluster_40 | OG0000385 | 0.25070199 | Yes | No | 154 |  9 | 0.05844156 |  4 | 0.02597403 | ⋯ | 146 | 0.9480519 |  8 | 0.05194805 | 146 | 0.9480519 |  8 | 0.05194805 | tick support        |  2 |\n",
       "| cluster_40 | OG0000143 | 0.15338398 | Yes | No | 282 | 11 | 0.03900709 | 11 | 0.03900709 | ⋯ | 232 | 0.8226950 | 10 | 0.03546099 | 232 | 0.8226950 | 10 | 0.03546099 | tick support        |  5 |\n",
       "| cluster_27 | OG0000079 | 0.08290626 | Yes | No | 394 | 55 | 0.13959391 | NA |         NA | ⋯ | 293 | 0.7436548 | 50 | 0.12690355 | 293 | 0.7436548 | 50 | 0.12690355 | tick support        | 28 |\n",
       "\n"
      ],
      "text/plain": [
       "   traitmapping_cluster traitmapping_orthogroup traitmapping_coefficient\n",
       "1  cluster_40           OG0011284               2.24534051              \n",
       "2  cluster_40           OG0008888               1.69892817              \n",
       "3  cluster_40           OG0001774               1.05408448              \n",
       "4  cluster_33           OG0008102               0.89907992              \n",
       "5  cluster_27           OG0002194               0.87213413              \n",
       "6  cluster_27           OG0000189               0.50428119              \n",
       "7  cluster_46           OG0000746               0.42872398              \n",
       "8  cluster_27           OG0000194               0.35313516              \n",
       "9  cluster_33           OG0000880               0.31400766              \n",
       "10 cluster_46           OG0001663               0.29542423              \n",
       "11 cluster_40           OG0000385               0.25070199              \n",
       "12 cluster_40           OG0000143               0.15338398              \n",
       "13 cluster_27           OG0000079               0.08290626              \n",
       "   traitmapping_signif_level traitmapping_signif_fdr num_proteins_in_orthogroup\n",
       "1  Yes                       No                        9                       \n",
       "2  Yes                       No                       16                       \n",
       "3  Yes                       No                       66                       \n",
       "4  Yes                       No                       24                       \n",
       "5  Yes                       No                       56                       \n",
       "6  Yes                       No                      240                       \n",
       "7  Yes                       No                      102                       \n",
       "8  Yes                       No                      237                       \n",
       "9  Yes                       No                      109                       \n",
       "10 Yes                       No                       64                       \n",
       "11 Yes                       No                      154                       \n",
       "12 Yes                       No                      282                       \n",
       "13 Yes                       No                      394                       \n",
       "   num_predicted_peptides fraction_of_orthogroup_with_predicted_peptide\n",
       "1   1                     0.11111111                                   \n",
       "2   1                     0.06250000                                   \n",
       "3  45                     0.68181818                                   \n",
       "4  18                     0.75000000                                   \n",
       "5   2                     0.03571429                                   \n",
       "6  21                     0.08750000                                   \n",
       "7   5                     0.04901961                                   \n",
       "8  23                     0.09704641                                   \n",
       "9  84                     0.77064220                                   \n",
       "10  6                     0.09375000                                   \n",
       "11  9                     0.05844156                                   \n",
       "12 11                     0.03900709                                   \n",
       "13 55                     0.13959391                                   \n",
       "   num_predicted_signal_peptides fraction_of_orthogroup_with_signal_peptide ⋯\n",
       "1   0                            0.00000000                                 ⋯\n",
       "2   7                            0.43750000                                 ⋯\n",
       "3  43                            0.65151515                                 ⋯\n",
       "4  17                            0.70833333                                 ⋯\n",
       "5   1                            0.01785714                                 ⋯\n",
       "6  22                            0.09166667                                 ⋯\n",
       "7   2                            0.01960784                                 ⋯\n",
       "8  NA                                    NA                                 ⋯\n",
       "9  57                            0.52293578                                 ⋯\n",
       "10  2                            0.03125000                                 ⋯\n",
       "11  4                            0.02597403                                 ⋯\n",
       "12 11                            0.03900709                                 ⋯\n",
       "13 NA                                    NA                                 ⋯\n",
       "   num_tick_proteins_in_orthogroup fraction_of_orthogroup_tick_proteins\n",
       "1    8                             0.8888889                           \n",
       "2   16                             1.0000000                           \n",
       "3   51                             0.7727273                           \n",
       "4   24                             1.0000000                           \n",
       "5   45                             0.8035714                           \n",
       "6  193                             0.8041667                           \n",
       "7   71                             0.6960784                           \n",
       "8  203                             0.8565401                           \n",
       "9  107                             0.9816514                           \n",
       "10  64                             1.0000000                           \n",
       "11 146                             0.9480519                           \n",
       "12 232                             0.8226950                           \n",
       "13 293                             0.7436548                           \n",
       "   num_predicted_peptides_from_tick\n",
       "1   1                              \n",
       "2   1                              \n",
       "3  38                              \n",
       "4  18                              \n",
       "5   2                              \n",
       "6  20                              \n",
       "7   4                              \n",
       "8  21                              \n",
       "9  82                              \n",
       "10  6                              \n",
       "11  8                              \n",
       "12 10                              \n",
       "13 50                              \n",
       "   fraction_of_orthogroup_with_predicted_tick_peptides\n",
       "1  0.11111111                                         \n",
       "2  0.06250000                                         \n",
       "3  0.57575758                                         \n",
       "4  0.75000000                                         \n",
       "5  0.03571429                                         \n",
       "6  0.08333333                                         \n",
       "7  0.03921569                                         \n",
       "8  0.08860759                                         \n",
       "9  0.75229358                                         \n",
       "10 0.09375000                                         \n",
       "11 0.05194805                                         \n",
       "12 0.03546099                                         \n",
       "13 0.12690355                                         \n",
       "   num_itchsuppsp_proteins_in_orthogroup\n",
       "1    8                                  \n",
       "2   16                                  \n",
       "3   52                                  \n",
       "4   24                                  \n",
       "5   45                                  \n",
       "6  195                                  \n",
       "7   72                                  \n",
       "8  203                                  \n",
       "9  107                                  \n",
       "10  64                                  \n",
       "11 146                                  \n",
       "12 232                                  \n",
       "13 293                                  \n",
       "   fraction_of_orthogroup_itchsuppsp_proteins\n",
       "1  0.8888889                                 \n",
       "2  1.0000000                                 \n",
       "3  0.7878788                                 \n",
       "4  1.0000000                                 \n",
       "5  0.8035714                                 \n",
       "6  0.8125000                                 \n",
       "7  0.7058824                                 \n",
       "8  0.8565401                                 \n",
       "9  0.9816514                                 \n",
       "10 1.0000000                                 \n",
       "11 0.9480519                                 \n",
       "12 0.8226950                                 \n",
       "13 0.7436548                                 \n",
       "   num_predicted_peptides_from_itchsuppsp\n",
       "1   1                                    \n",
       "2   1                                    \n",
       "3  39                                    \n",
       "4  18                                    \n",
       "5   2                                    \n",
       "6  20                                    \n",
       "7   4                                    \n",
       "8  21                                    \n",
       "9  82                                    \n",
       "10  6                                    \n",
       "11  8                                    \n",
       "12 10                                    \n",
       "13 50                                    \n",
       "   fraction_of_orthogroup_with_predicted_itchsuppsp_peptides\n",
       "1  0.11111111                                               \n",
       "2  0.06250000                                               \n",
       "3  0.59090909                                               \n",
       "4  0.75000000                                               \n",
       "5  0.03571429                                               \n",
       "6  0.08333333                                               \n",
       "7  0.03921569                                               \n",
       "8  0.08860759                                               \n",
       "9  0.75229358                                               \n",
       "10 0.09375000                                               \n",
       "11 0.05194805                                               \n",
       "12 0.03546099                                               \n",
       "13 0.12690355                                               \n",
       "   type_of_itch_suppression_evidence num_predicted_peptides_with_sg_blast_hit\n",
       "1  tick support                       1                                      \n",
       "2  tick support                       1                                      \n",
       "3  chelicerate support               34                                      \n",
       "4  tick support                      13                                      \n",
       "5  tick support                       1                                      \n",
       "6  tick support                      15                                      \n",
       "7  tick support                       2                                      \n",
       "8  tick support                      10                                      \n",
       "9  tick support                      30                                      \n",
       "10 tick support                       4                                      \n",
       "11 tick support                       2                                      \n",
       "12 tick support                       5                                      \n",
       "13 tick support                      28                                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "orthogroup_peptide_summary_filtered <- orthogroup_peptide_summary %>%\n",
    "  # filter out orthogroups that had no predicted peptides\n",
    "  filter(num_predicted_peptides > 0) %>%\n",
    "  # filter out orthogroups that had no evidence of being expressed in salivary gland\n",
    "  filter(num_predicted_peptides_with_sg_blast_hit > 0) %>%\n",
    "  arrange(desc(traitmapping_coefficient))\n",
    "\n",
    "orthogroup_peptide_summary_filtered "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8f540b81-bfd4-4d93-bf81-2d158ba95d11",
   "metadata": {},
   "outputs": [],
   "source": [
    "write_tsv(orthogroup_peptide_summary_filtered, \"outputs/notebooks/20240626_orthogroup_peptide_summary.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72b1c176-a557-4f20-ae74-b3ed2cdb1913",
   "metadata": {},
   "source": [
    "We made a data frame that summarizes our ToT peptigate results.\n",
    "See a description of the columns below.\n",
    "\n",
    "* **num_proteins_in_orthogroup**: The number of proteins in the orthogroups assigned by NovelTree.\n",
    "* Number of proteins in orthogroup with peptide prediction\n",
    "    * **num_predicted_peptides**: The number of predicted peptides for that orthogroup. For example, if the orthogroup had 10 proteins, 3 may have had peptides predicted from them.\n",
    "    * **fraction_of_orthogroup_with_predicted_peptide**: Fraction of proteins that had peptides predicted for the orthogroup.\n",
    "* Number of proteins in orthogroup from ticks (order *Ixodida*), and how many had peptide predictions.\n",
    "    * **num_tick_proteins_in_orthogroup**: The number of proteins in the orthogroup that were from tick species (order *Ixodida*).\n",
    "    * **fraction_of_orthogroup_tick_proteins**: The fraction of proteins in the orthogroup that were from tick species (order *Ixodida*).\n",
    "    * **num_predicted_peptides_from_tick**: The number of peptides that were predicted for the orthogroup that were from a tick species (order *Ixodida*).\n",
    "    * **fraction_of_orthogroup_with_predicted_tick_peptides**: Fraction of proteins that had peptides predicted from a tick species (order *Ixodida*) for the orthogroup.\n",
    "* Number of proteins in orthogroup from chelicerates that suppress itch. `itchsuppsp` stands for \"itch suppression species\"\n",
    "    * **num_itchsuppsp_proteins_in_orthogroup**: Number of proteins in the orthogroup from itch-suppressing species. This include tick species (order *Ixodida*) as well as two non-tick itch suppressors, *Leptotrombidium deliense* and *Sarcoptes scabiei*.\n",
    "    * **fraction_of_orthogroup_itchsuppsp_proteins**: Fraction of proteins in the orthogroup from itch-suppressing species.\n",
    "    * **num_predicted_peptides_from_itchsuppsp**: Number of peptides predicted from itch-suppressing species in the orthogroup.\n",
    "    * **fraction_of_orthogroup_with_predicted_itchsuppsp_peptides**: Fraction of peptides predicted from itch-suppressiong species in the orthogroup.\n",
    "* Calculate whether there is tick & non-tick evidence of itch suppression\n",
    "    * **type_of_itch_suppression_evidence**: Either `chelicerate support`, `tick support`, or `no support`. `Chelicerate support` means that the orthogroup was associated with itch suppression and that there were peptides predicted from both ticks and non-tick itch suppressors that are chelicerates. We think these probably show some of the best evolutionary support for itch suppression but please still think critically about these predictions. `Tick support` means that the orthogroup was associated with itch suppression and peptides were only predicted from tick species. `No support` means that there were no peptides predicted from tick or from the two non-tick itch suppressors.\n",
    "* Determine whether any member of the orthogroup had a BLAST hit against peptides predicted from the tick salivary gland transcriptomes\n",
    "    * **num_predicted_peptides_with_sg_blast_hit**: Number of predicted peptides that had hits to peptides predicted from tick salivary gland transcriptomes. We only kept orthogroups that had at least one hit to a tick salivary gland transcriptome peptide. **Note, we need to check the *Amblyomma americanum* hits** -- if these hits are to transcripts that start with the word \"Transcript\", these hits are from a whole tick transcriptome, not just a salivary gland, so we need to double check that they are expressed in the salivary gland. (The salivary gland transcripts made up over 1/3 of the transcriptome so likelihood is high that they do hit salivary gland transcripts)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "808fc82b-c470-4e76-a0bc-38a1f596c933",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.3.3 (2024-02-29)\n",
       "Platform: x86_64-apple-darwin13.4.0 (64-bit)\n",
       "Running under: macOS Big Sur ... 10.16\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /Users/taylorreiter/miniconda3/envs/tidyjupyter/lib/libopenblasp-r0.3.26.dylib;  LAPACK version 3.12.0\n",
       "\n",
       "locale:\n",
       "[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8\n",
       "\n",
       "time zone: America/New_York\n",
       "tzcode source: internal\n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       " [1] UpSetR_1.4.0    lubridate_1.9.3 forcats_1.0.0   stringr_1.5.1  \n",
       " [5] dplyr_1.1.4     purrr_1.0.2     readr_2.1.5     tidyr_1.3.1    \n",
       " [9] tibble_3.2.1    ggplot2_3.5.0   tidyverse_2.0.0\n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] bit_4.0.5        gtable_0.3.4     jsonlite_1.8.8   compiler_4.3.3  \n",
       " [5] crayon_1.5.2     Rcpp_1.0.12      tidyselect_1.2.0 IRdisplay_1.1   \n",
       " [9] parallel_4.3.3   gridExtra_2.3    scales_1.3.0     uuid_1.2-0      \n",
       "[13] fastmap_1.1.1    IRkernel_1.3.2   plyr_1.8.9       R6_2.5.1        \n",
       "[17] generics_0.1.3   munsell_0.5.1    pillar_1.9.0     tzdb_0.4.0      \n",
       "[21] rlang_1.1.3      utf8_1.2.4       stringi_1.8.3    repr_1.1.7      \n",
       "[25] bit64_4.0.5      timechange_0.3.0 cli_3.6.2        withr_3.0.0     \n",
       "[29] magrittr_2.0.3   digest_0.6.35    grid_4.3.3       vroom_1.6.5     \n",
       "[33] base64enc_0.1-3  hms_1.1.3        pbdZMQ_0.3-11    lifecycle_1.0.4 \n",
       "[37] vctrs_0.6.5      evaluate_0.23    glue_1.7.0       fansi_1.0.6     \n",
       "[41] colorspace_2.1-0 tools_4.3.3      pkgconfig_2.0.3  htmltools_0.5.8 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6549d3f-77bd-4dbb-97c5-f331d905b3f2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
